calibre-web-automated-downloader/src/models.rs

```diff
@@ -0,0 +1,217 @@
+    pub info: Option<HashMap<String, Vec<String>>>,
+
+    /// e.g. a list of direct download URLs
+    pub download_urls: Vec<String>,
+}
+
+impl BookInfo {
+    /// Convenience constructor if desired
+    pub fn new(id: &str, title: &str) -> Self {
+        Self {
+            id: id.to_owned(),
+            title: title.to_owned(),
+            preview: None,
+            author: None,
+            publisher: None,
+            year: None,
+            language: None,
+            format: None,
+            size: None,
+            info: None,
+            download_urls: vec![],
+        }
+    }
+}
+
+/// Thread-safe book queue manager
+#[derive(Debug)]
+pub struct BookQueue {
+    queue: HashSet<String>,
+    status: HashMap<String, QueueStatus>,
+    book_data: HashMap<String, BookInfo>,
+    status_timestamps: HashMap<String, Instant>,
+    status_timeout: Duration,
+}
+
+impl BookQueue {
+    /// Create a new empty queue with a default timeout (from CONFIG).
+    pub fn new() -> Self {
+        let timeout_secs = CONFIG.status_timeout;
+        BookQueue {
+            queue: HashSet::new(),
+            status: HashMap::new(),
+            book_data: HashMap::new(),
+            status_timestamps: HashMap::new(),
+            status_timeout: Duration::from_secs(timeout_secs),
+        }
+    }
+
+    /// Internal helper to update the status + timestamp for a book ID
+    fn update_status_internal(&mut self, book_id: &str, status: QueueStatus) {
+        self.status.insert(book_id.to_string(), status);
+        self.status_timestamps
+            .insert(book_id.to_string(), Instant::now());
+    }
+
+    /// Add a book to the queue
+    pub fn add(&mut self, book_id: &str, book_data: BookInfo) {
+        self.queue.insert(book_id.to_string());
+        self.book_data.insert(book_id.to_string(), book_data);
+        self.update_status_internal(book_id, QueueStatus::Queued);
+    }
+
+    /// Get the next book in the queue (if any)
+    pub fn get_next(&mut self) -> Option<String> {
+        if let Some(book_id) = self.queue.iter().next().cloned() {
+            self.queue.remove(&book_id);
+            Some(book_id)
+        } else {
+            None
+        }
+    }
+
+    /// Update status (public method)
+    pub fn update_status(&mut self, book_id: &str, status: QueueStatus) {
+        if self.status.contains_key(book_id) {
+            self.update_status_internal(book_id, status);
+        }
+    }
+
+    /// Return the current status of all books, grouped by QueueStatus.
+    ///
+    /// In Python: Dict[QueueStatus, Dict[str, BookInfo]]
+    /// In Rust: HashMap<QueueStatus, HashMap<String, BookInfo>>
+    pub fn get_status(&mut self) -> HashMap<QueueStatus, HashMap<String, BookInfo>> {
+        // First refresh to remove stale/done items
+        self.refresh();
+
+        let mut result = HashMap::<QueueStatus, HashMap<String, BookInfo>>::new();
+
+        // Initialize an empty HashMap for each status variant
```

Comment:
Hardcode this and initialize the outer hashmap with these keys, as the status variants are already hardcoded.